examples/hybrid/sphere/baroclinic_wave_utilities.jl

```diff
@@ -606,22 +603,42 @@ function constant_T_saturated_surface_coefs(
     Ch,
     params,
 )
+    # parameters
     thermo_params = CAP.thermodynamics_params(params)
+    sf_params = CAP.surface_fluxes_params(params)
+
+    # get the near-surface thermal state
     T_int = TD.air_temperature(thermo_params, ts_int)
     Rm_int = TD.gas_constant_air(thermo_params, ts_int)
     ρ_sfc =
         TD.air_density(thermo_params, ts_int) *
         (T_sfc / T_int)^(TD.cv_m(thermo_params, ts_int) / Rm_int)
+
     q_sfc =
         TD.q_vap_saturation_generic(thermo_params, T_sfc, ρ_sfc, TD.Liquid())
     ts_sfc = TD.PhaseEquil_ρTq(thermo_params, ρ_sfc, T_sfc, q_sfc)
-    return SF.Coefficients{FT}(;
+
+    # wrap state values
+    sc = SF.Coefficients{FT}(;
         state_in = SF.InteriorValues(z_int, (uₕ_int.u, uₕ_int.v), ts_int),
         state_sfc = SF.SurfaceValues(z_sfc, (FT(0), FT(0)), ts_sfc),
-        Cd,
-        Ch,
         z0m = FT(0),
         z0b = FT(0),
+        Cd = Cd, #FT(0.001),
+        Ch = Ch, #FT(0.0001),
+    )
+
+    # calculate all fluxes
+    tsf = SF.surface_conditions(sf_params, sc, SF.FVScheme())
+
+    E = SF.evaporation(sc, sf_params, tsf.Ch)
+
+    return (;
+        shf = tsf.shf,
+        lhf = tsf.lhf,
+        E = E,
+        ρτxz = tsf.ρτxz,
+        ρτyz = tsf.ρτyz,
     )
```

Comment:
I think that this is all going to be allocating

Reply:
Should I put them in the cache then? Maybe it would be better to have a pair programming session if you have time?

Reply:
Let's see how the perf summary looks. It shouldn't be that complicated, though